docs/serverdev/index.rst

```diff
@@ -0,0 +1,130 @@
+Contributing to Sentry
```

Comment:
Added a little bit of that in the first sentence, because ``Contributing to Sentry Server`` wraps around the sidebar. If you have a better title, lmk. This very well could turn into a generic contributing doc but I wanted to nail out the server-side one first.